### PR TITLE
feat(HACBS-1050): Add codecov to integration-service

### PR DESCRIPTION
**Pull request by srivickynesh**
Jira : [HACBS-1050](https://issues.redhat.com/browse/HACBS-1050)

Signed-off-by: Sri Vignesh <sselvan@redhat.com>

### CODE REVIEW

File: codecov.yml [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+        # Allows coverage to drop by a 3% when compared against the base commit.
+        project:
+            default:
+                target: auto
+                threshold: 3%
+        # Sets the expected status for `codecov/patch` check.
+        # We set this to be only informational hence it does not cause the check to fail.
+        patch:
+            default:
+                informational: true
```
Comment (Dannyb48) [inline codecov.yml:13]: Do we not want to block? or maybe I don't know enough about this option?

File: codecov.yml [hunk 2]
```diff
@@ -0,0 +1,13 @@
+# Documentation: https://docs.codecov.io/docs/codecov-yaml
```
Comment (dirgim) [inline codecov.yml:1]: Can we also add the setting to ignore the generated `api/v1alpha1/zz_generated.deepcopy.go` file? Example from the release-service: https://github.com/redhat-appstudio/release-service/blob/main/codecov.yml#L2-L3.

Comment (srivickynesh) [reply codecov.yml:1]: okay, will add.

File: codecov.yml (hunk 1 above)
Comment (dirgim) [reply codecov.yml:13]: I think that we do not want to block on codecov because we've been hearing that codecov sometimes behaves weirdly for some PRs.

Comment (srivickynesh) [reply codecov.yml:13]: Sometimes the difference calculated by codecov is 100% from the previous commit.
Informational only - this way we still have the info but it does not fail if the commit in the same PR has a difference of 100%.
Checking the coverage diff between commits (patch) as it tends to change when rebasing the code.

File: codecov.yml (hunk 2 above)
Comment (dirgim) [reply codecov.yml:1]: thanks!

**Review by dirgim — APPROVED** [review]
LGTM

File: codecov.yml (hunk 1 above)
Comment (Dannyb48) [reply codecov.yml:17]: Ack

**Review by Dannyb48 — APPROVED** [review]
LGTM

**Pull request merged**